packages/typescript-sdk/src/necessary-middlewares/ctp_middlware.ts

```diff
@@ -0,0 +1,47 @@
+import { Middleware, MiddlewareArg } from "../gen/base/common-types";
+
+export function middwareFromCtpClient(client:any): Middleware {
+  return async (midlllewareArg: MiddlewareArg) =>
+    handleRequest(client, midlllewareArg);
+}
+
+async function handleRequest(
+  client:any,
+  middlwareArg: MiddlewareArg
+): Promise<MiddlewareArg> {
+  const { request } = middlwareArg;
+
+  const modifiedRequest = {
+    ...request,
+    uri: removeBaseUrl(request.uri)
+  };
+
+  try {
+    const result = await client.execute(modifiedRequest);
+    const { uri, error, ...response } = result;
+
+    return middlwareArg.next({
+      ...middlwareArg,
+      response,
+      error
+    });
+  } catch (error) {
+    return middlwareArg.next({
+      ...middlwareArg,
+      error
+    });
+  }
+}
+
+function removeBaseUrl(url: string) {
+  var baseUrlPattern = /^https?:\/\/[a-z\:0-9.]+/;
+  var result = "";
+  var match = baseUrlPattern.exec(url);
+  if (match != null) {
```

Comment:
any special reason we choose to not strictly check this one with `!==` ?

Reply:
nope, i replaced it with `if (match) `

Reply:
Coolio, then maybe we could just make it into a oneliner?
`const result = match ? match[0] : ""`